lib/pbench/server/api/resources/query_apis/controllers_list.py

```diff
@@ -55,19 +57,24 @@ def assemble(self, json_data: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
                 "access": "public", "access": "private", or "access": "all" to
                 include both private and public data.
 
-            "start" and "end" are datetime objects representing a set of Elasticsearch
-                run document indices in which to search.
+            "start" and "end" are datetime objects: these are used to select a
+                set of YYYY-mm sequenced Elasticsearch run indices to search,
+                as well as a more fine-grained comparison against the embedded
+                document "@timestamp" data.
+
+        context: Context passed from preprocess method: not used here.
         """
-        user = json_data["user"]
+        user = json_data.get("user")
         start = json_data["start"]
         end = json_data["end"]
 
-        # We need to pass string dates as part of the Elasticsearch query; we
-        # use the unconverted strings passed by the caller rather than the
-        # adjusted and normalized datetime objects for this.
-        start_arg = f"{start:%Y-%m}"
-        end_arg = f"{end:%Y-%m}"
-
+        # We need to pass string dates as part of the Elasticsearch query. The
+        # _gen_month_range constructs a set of Elasticsearch sequences index
+        # names based on the YYYY-MM granularity of those dates; for this query
+        # we also use the original dates to search for documents within those
+        # indices.
+        start_arg = start.strftime("%Y-%m-%d")
+        end_arg = end.strftime("%Y-%m-%d")
```

Comment:
OK, this should go back to the f-string format. Somewhere early in this process I'd thought I could just use the full date/time string; but Elasticsearch barfs on that, and I backed off to just YYYY-MM-DD and added `strftime` without really thinking about it. Should be just `f"{start:%Y-%m-%d}"`

Reply:
> Elasticsearch barfs on that

IIRC, ES has pretty flexible date string processing, you just have to set the mapping appropriately.   Perhaps that's just for ingestion...perhaps not for queries....  😕

Reply:
I'm not going to change the Elasticsearch field mapping for this (assuming that would even be possible), and if giving Elasticsearch this little gift appeases it, that's fine.